Python/Product/Debugger/Debugger/DebugEngine/AD7Engine.cs

```diff
@@ -1436,9 +1436,13 @@ public sealed class AD7Engine : IDebugEngine2, IDebugEngineLaunch2, IDebugProgra
         }
 
         private void OnDebuggerOutput(object sender, OutputEventArgs e) {
-            AD7Thread thread;
-            if (!_threads.TryGetValue(e.Thread, out thread)) {
-                _threads[e.Thread] = thread = new AD7Thread(this, e.Thread);
+            // Output from debug REPL code execution may be run on debugger
+            // event handling thread, and e.Thread will be null.
+            AD7Thread thread = null;
+            if (e.Thread != null) {
+                if (!_threads.TryGetValue(e.Thread, out thread)) {
```

Comment:
It's not protected anywhere else, but that doesn't mean that's correct ;) That could be the cause of the multithreading debugging issues we have trouble reproducing.